src/content-proxy.js

```diff
@@ -47,20 +63,27 @@ async function contentProxy(opts) {
   url.searchParams.append('rid', options.requestId);
 
   log.info(`Fetching content from: ${url.href}`);
-  const response = await fetch(url.href, getFetchOptions(options));
-  const body = await response.text();
-  if (response.ok) {
+  const resp = await fetch(url.href, getFetchOptions(options));
+  const body = await resp.buffer();
+  if (resp.ok) {
+    const headers = {};
+    PASSTHROUGH_HEADERS.forEach((name) => {
+      const value = resp.headers.get(name);
+      if (value) {
+        headers[name] = value;
+      }
+    });
     return new Response(body, {
```

Comment:
why are you recreating a response and not just return a plain object?
the `Response` initializer might cause extra overhead.